styx-common/src/test/java/com/spotify/styx/util/ShardedCounterTest.java

```diff
@@ -374,7 +377,8 @@ public void shouldFailDecrementingEmptyShard() throws IOException {
 
   @Test(expected = ShardNotFoundException.class)
   public void shouldThrowExceptionOnUninitializedShards() {
-    when(counterSnapshotFactory.create(COUNTER_ID1)).thenReturn(new ShardedCounter.Snapshot(COUNTER_ID1, 100, new HashMap<>()));
+    doReturn(new ShardedCounter.Snapshot(COUNTER_ID1, 100, new HashMap<>()))
```

Comment:
Is this change necessary?

Reply:
Sure but it is not storage here but that counterSnapshotFactory.

Reply:
Yes but counterSnapshotFactory is using the spy storage.

Reply:
OK then. I missed that connection.

Reply:
Sorry I meant that `counterSnapshotFactory` is a spy and that breaks `when(...).then(...)`.

Reply:
Ah, so it never worked before. That's interesting.

Reply:
It has worked before (by accident, luck?), but it broke in this PR because now we inject a mock Stats into AggregateStorage. And then in the `when(...)` the _real_ `counterSnapshotFactory.create` was called, which it the mocked `Stats`, which broke Mockito. And that's the gist of the problem. When doing `when(...)` on a spy, the _real_ method gets called immediately then and there, and we don't want that, that breaks things.

Reply:
Yeah. It was never the right way working with `spy`, and fortunately this change surfaced it.